cpp/src/parquet/arrow/reader.h

```diff
@@ -141,26 +141,19 @@ class PARQUET_EXPORT FileReader {
 
   /// \brief Read column as a whole into a chunked array.
   ///
-  /// The indicated column index is relative to the schema
+  /// The index i refers the index of the top level schema field, which may
```

Comment:
I think maybe this is wrong. `ReadColumn` will like:

```
  ///   foo.bar
  /// 0 foo.bar.baz
  /// 1 foo.qux
  /// 2 foo2
  /// 3 foo3
```

It would only works for leaf type. And I think previous `ReadSchemaField` has bug for this

Reply:
Oh you're right, this will call

```
Status GetReader(const SchemaField& field, const std::shared_ptr<Field>& arrow_field,
                 const std::shared_ptr<ReaderContext>& ctx,
                 std::unique_ptr<ColumnReaderImpl>* out)
```

Which build a reader for complex type.

Reply:
I get this confused with `ReadTable`, it will only read the leaf-node, sorry for that